packages/trader/src/AppV2/Components/TradeParameters/Stake/stake.tsx

```diff
@@ -1,37 +1,228 @@
+                values={{
+                    currency: getCurrencyDisplayCode(currency),
+                    min_stake: FormatUtils.formatMoney(+details.min_stake),
+                    max_stake: FormatUtils.formatMoney(+details.max_stake),
+                }}
+            />
+        ));
+
+    const handleOnChange = (e: { target: { name: string; value: string } }) => {
+        setShouldShowError(e.target.value !== '');
+        onChange({ target: { name: 'amount', value: e.target.value } });
+    };
+
+    const onClose = (is_saved = false) => {
+        if (is_open) {
+            if (!is_saved) {
+                onChange({ target: { name: 'amount', value: v2_params_initial_values.stake } });
+            }
+            if (v2_params_initial_values.stake !== amount) {
+                setV2ParamsInitialValues({ value: amount, name: 'stake' });
+            }
+            setIsOpen(false);
+        }
+    };
 
-    if (basis === BASIS.PAYOUT) return null;
     return (
-        <TextField
-            variant='fill'
-            readOnly
-            label={<Localize i18n_default_text='Stake' key={`stake${is_minimized ? '-minimized' : ''}`} />}
-            value={`${amount} ${getCurrencyDisplayCode(currency)}`}
-            className={clsx('trade-params__option', is_minimized && 'trade-params__option--minimized')}
-        />
+        <>
+            <TextField
+                variant='fill'
+                readOnly
+                label={<Localize i18n_default_text='Stake' key={`stake${is_minimized ? '-minimized' : ''}`} />}
+                onClick={() => setIsOpen(true)}
+                value={`${v2_params_initial_values?.stake ?? amount} ${getCurrencyDisplayCode(currency)}`}
+                className={clsx('trade-params__option', is_minimized && 'trade-params__option--minimized')}
+            />
+            <ActionSheet.Root isOpen={is_open} onClose={() => onClose(false)} position='left' expandable={false}>
+                <ActionSheet.Portal shouldCloseOnDrag>
+                    <ActionSheet.Header title={<Localize i18n_default_text='Stake' />} />
+                    <ActionSheet.Content className='stake-content'>
+                        <TextFieldWithSteppers
+                            allowDecimals
+                            decimals={getDecimalPlaces(currency)}
+                            data-testid='dt_input_with_steppers'
+                            message={getInputMessage()}
+                            minusDisabled={Number(amount) - 1 <= 0}
+                            name='amount'
+                            onChange={handleOnChange}
+                            placeholder={localize('Amount')}
+                            regex={/[^0-9.,]/g}
+                            status={should_show_error && stake_error ? 'error' : 'neutral'}
+                            textAlignment='center'
+                            type='number'
```

Comment:
Does it allow comma/dot? I thought type number does not

Reply:
it allows dot but not the comma, same like in production :)

Reply:
But the problem is that some numeric keyboard have comma instead of dot ( Will such users be able to edit value?